app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -2,10 +2,34 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   include PreferencesController
 
   require_authentication :all, scopes: [:project]
+  extra_schema_actions :update_settings
   schema_type :json_schema
+  before_action :validate_records, only: [:update_settings]
+
+  def update_settings
+    upp = UserProjectPreference.find_or_initialize_by(project_id: params_for[:project_id],
```

Comment:
Good question, I wasn't sure. Since the user ids will be presumably coming from classifications (in gravityspy's case, at least) then the UPPs will exist (...right?). In general, though, it does seem reasonable to generally require the UPP to exist before you can mess with it via this route. Changed to a .where, which should bubble up a RecordNotFound like the others (project/user).